spec/app/domain/services/update_course_exercise_exclusions/service_spec.rb

```diff
@@ -0,0 +1,155 @@
+  ) }
+
+  let(:action_many) { service.process(
+    update_uuid:      given_update_uuid,
+    sequence_number:  given_sequence_number,
+    course_uuid:      given_course_uuid,
+    exclusions:       given_many_exclusions.exclusions
+  ) }
+
+  let(:action_repeat) { service.process(
+    update_uuid:      given_update_uuid,
+    sequence_number:  given_sequence_number,
+    course_uuid:      given_course_uuid,
+    exclusions:       given_some_exclusions.exclusions
+  ) }
+
+  let(:given_none_exclusions) {
+    build_stubbed(:exercise_exclusions, exclusions_count: number_of_exclusions_none,
+                              exclusions_any_count: number_of_exclusions_none,
+                              exclusions_specific_count: number_of_exclusions_none)
+  }
+
+  let(:given_some_exclusions) {
+    number_of_exclusions_any = Kernel::rand(number_of_exclusions_some)
+    number_of_exclusions_specific = number_of_exclusions_some - number_of_exclusions_any
+
+    build_stubbed(:exercise_exclusions, exclusions_count: number_of_exclusions_some,
+                              exclusions_any_count: number_of_exclusions_any,
+                              exclusions_specific_count: number_of_exclusions_specific)
+  }
+
+  let(:given_many_exclusions) {
+    number_of_exclusions_any = Kernel::rand(number_of_exclusions_many)
+    number_of_exclusions_specific = number_of_exclusions_many - number_of_exclusions_any
+
+    build_stubbed(:exercise_exclusions, exclusions_count: number_of_exclusions_many,
+                              exclusions_any_count: number_of_exclusions_any,
+                              exclusions_specific_count: number_of_exclusions_specific)
+  }
+
+  let(:given_update_uuid)           { SecureRandom.uuid.to_s }
+  let(:given_sequence_number)       { Kernel::rand(10) }
+  let(:given_course_uuid)           { SecureRandom.uuid.to_s }
+
+  let(:number_of_exclusions_none)   { 0 }
+  let(:number_of_exclusions_some)   { 10 }
+  let(:number_of_exclusions_many)   { 100 }
+
+  context "when a previously non-existing Course uuid is given" do
+    it "raises error" do
+      expect{action_some}.to raise_error(Errors::AppUnprocessableError)
+    end
+  end
+
+  context "when previously existing Course uuid is given" do
+    before(:each) do
+      create(:course, uuid: given_course_uuid)
+    end
+
+    context "and previously existing request is given" do
+      before(:each) do
+        action_repeat
+      end
+
+      context "with no exclusions" do
+        it "raises error" do
+          expect{action_none}.to raise_error(ActiveRecord::RecordNotUnique)
+        end
+      end
+
+      context "with some exclusions" do
+        it "raises error" do
+          expect{action_some}.to raise_error(ActiveRecord::RecordNotUnique)
+        end
+      end
+
+      context "with many exclusions" do
+        it "raises error" do
+          expect{action_many}.to raise_error(ActiveRecord::RecordNotUnique)
+        end
+      end
+    end
+
```

Comment:
These specs make repeated requests non-idempotent.  They must be handed the same way as the originals, returning the same results.